### ISSUE TITLE
Constructor to global variable in converter generates illegal c code

### ISSUE THREAD

**Issue by Tarmean** [nim-lang/Nim#4594]
This is a tad difficult to explain so I am going to give examples. This generates illegal c code:

``` nim
type
  Inner =  object
    foo: string
converter toString(i: int): string =
  var g{.global.} = Inner(foo: $i)
var s: string = 4
```

> src\nimcache\converter_test.c: In function 'converter_testInit000':
> src\nimcache\converter_test.c:354:30: error: 'i0' undeclared (first use in this function)  
> LOC100001.foo = nimIntToStr(i0);
>                               ^
> src\nimcache\converter_test.c:354:30: note: each undeclared identifier is reported only once for each  function it appears in

In contrast this does compile and work as expected:

``` nim
type
  Inner =  object
    foo: string
converter toString(i: int): string =
  var g{.global.} : Inner
   g.foo = $i
var s: string = 4
```

When the string is constant, say Inner(foo: "bar"), or the converter is invoked explicitely the program also works.

**Comment by Tarmean** [comment]
I am not sure whether this is related or should be in a new issue, but:

```nim
converter conv(s: var string): int =
  echo s
var 
    a = "a"
    b: int = a
```

crashes with a segmentation fault when s is dereferenced. This also only happens if conv is called implicitely.

**Comment by cooldome** [comment]
I have tested this one, there is nothing converter specific. The following examples fails the same way:
```nim
type
  Inner =  object
    foo: string
proc toString(i: int): string =
  var g{.global.} : Inner
  g.foo = $i
var s = toString(4)
```

Please remove `converters` label. Makes sense to update the title as well

**Comment by bung87** [comment]
cannot reproduce in current devel d8d08324d3f761fcef60433a1c35b97c4874be20

**Comment by ringabout** [comment]
The first shouldn't compile because it uses local varibles. The second examples works
